### PR TITLE
Use Subprocess.env instead of injecting env into bash

### PR DESCRIPTION
**Pull request by ozdanborne**
## Description

Use `subprocess.Env` instead of `bash -c 'ENV ...'`

:taco: 

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: test_utils/utils.go [hunk 1]
```diff
@@ -209,20 +215,31 @@ func RunCNIPluginWithId(netconf, k8sName, ip, containerId string, targetNs ns.Ne
 
 	// Set up the env for running the CNI plugin
 	//TODO pass in the env properly
-	var k8s_env = ""
+	var k8sEnv = ""
```
Comment (gunjan5) [inline test_utils/utils.go]: 👍

File: test_utils/utils.go [hunk 2]
```diff
@@ -209,20 +215,31 @@ func RunCNIPluginWithId(netconf, k8sName, ip, containerId string, targetNs ns.Ne
 
 	// Set up the env for running the CNI plugin
 	//TODO pass in the env properly
```
Comment (gunjan5) [inline test_utils/utils.go]: this comment can go now

File: test_utils/utils.go [hunk 3]
```diff
@@ -209,20 +215,31 @@ func RunCNIPluginWithId(netconf, k8sName, ip, containerId string, targetNs ns.Ne
...
+		"CNI_PATH=dist",
+		fmt.Sprintf("CNI_CONTAINERID=%s", containerId),
+		fmt.Sprintf("CNI_NETNS=%s", targetNs.Path()),
+		k8sEnv,
+	}
+
+	log.Debug(env)
 
 	// Run the CNI plugin passing in the supplied netconf
 	//TODO - Get rid of this PLUGIN thing and use netconf instead
```
Comment (gunjan5) [inline test_utils/utils.go]: can you add a space after the `//` while you're at it :)

File: test_utils/utils.go (hunk 1 above)
Comment (gunjan5) [inline test_utils/utils.go]: Can you make this `k8sEnv := ""`? That is more readable

File: test_utils/utils.go [hunk 4]
```diff
@@ -300,16 +317,26 @@ func DeleteContainerWithId(netconf, netnspath, name, containerId string) (exitCo
 	if containerId != "" {
 		container_id = containerId
 	}
-	var k8s_env = ""
+	var k8sEnv = ""
```
Comment (gunjan5) [inline test_utils/utils.go]: same here

File: test_utils/utils.go [hunk 5]
```diff
@@ -209,20 +215,31 @@ func RunCNIPluginWithId(netconf, k8sName, ip, containerId string, targetNs ns.Ne
...
+	env := []string{
+		"CNI_COMMAND=ADD",
+		"CNI_IFNAME=eth0",
+		"CNI_PATH=dist",
+		fmt.Sprintf("CNI_CONTAINERID=%s", containerId),
+		fmt.Sprintf("CNI_NETNS=%s", targetNs.Path()),
+		k8sEnv,
+	}
+
+	log.Debug(env)
```
Comment (gunjan5) [inline test_utils/utils.go]: should prolly add a message "Using env vars: %v", env ... something like that

File: test_utils/utils.go [hunk 6]
```diff
@@ -209,20 +215,31 @@ func RunCNIPluginWithId(netconf, k8sName, ip, containerId string, targetNs ns.Ne
...
 
 	// Run the CNI plugin passing in the supplied netconf
 	//TODO - Get rid of this PLUGIN thing and use netconf instead
-	subProcess := exec.Command("bash", "-c", fmt.Sprintf("%s dist/%s", cni_env, os.Getenv("PLUGIN")), netconf)
+	subProcess := exec.Command(fmt.Sprintf("dist/%s", os.Getenv("PLUGIN")), netconf)
+	subProcess.Env = env
 	stdin, err := subProcess.StdinPipe()
 	if err != nil {
 		panic("some error found")
```
Comment (gunjan5) [inline test_utils/utils.go]: panic... ouch. Can we make this an assertion?

Comment (ozdanborne) [inline test_utils/utils.go]: 10 panics in this file. Lets save it for the next one.

**Pull request merged**